File: result/translate/transcoder/java2cpp/c++/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE.cpp
int f_gold ( int notes [ ] , int n ) {
  int five_count = 0 ;
  int ten_count = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( notes [ i ] == 5 ) {
      five_count ++ ;
    }
    else if ( notes [ i ] == 10 ) {
      if ( five_count > 0 ) {
        five_count -- ;
        ten_count ++ ;
      }
      else return 0 ;
    }
    else {
      if ( five_count > 0 && ten_count > 0 ) {
        five_count -- ;
        ten_count -- ;
      }
      else if ( five_count >= 3 ) five_count -= 3 ;
      else return 0 ;
    }
  }
  return 1 ;
}

File: result/translate/starchat/python2java/java/CHECK_IF_X_CAN_GIVE_CHANGE_TO_EVERY_PERSON_IN_THE_QUEUE.java
public static boolean f_gold ( int[] notes, int n ) {
        int fiveCount = 0, tenCount = 0;
        for ( int i = 0; i < n; i++ ) {
            if ( notes[i] == 5 ) fiveCount++;
            else if ( notes[i] == 10 ) {
                if ( fiveCount > 0 ) {
                    fiveCount--;
                    tenCount++;
                } else return false;
            } else {
                if ( fiveCount > 0 && tenCount > 0 ) {
                    fiveCount--;
                    tenCount--;
                } else if ( fiveCount >= 3 ) fiveCount -= 3;
                else return false;
            }
        }
        return true;
    }